plugins/texteditor.koplugin/main.lua

```diff
@@ -0,0 +1,160 @@
+local ConfirmBox = require("ui/widget/confirmbox")
+local DataStorage = require("datastorage")
+local InfoMessage = require("ui/widget/infomessage")
+local InputDialog = require("ui/widget/inputdialog")
+local PathChooser = require("ui/widget/pathchooser")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local logger = require("logger")
+local util = require("ffi/util")
+local _ = require("gettext")
+local Screen = require("device").screen
+
+
+local TextEditor = WidgetContainer:new{
+    name = "text_editor",
+}
+
+function TextEditor:init()
+    self.ui.menu:registerToMainMenu(self)
+end
+
+function TextEditor:start(file_path)
+    local FileManager = require("apps/filemanager/filemanager")
+    FileManager:onClose()
+    if not file_path then
+        self.context = ""
+        self.file_path = ""
+    else
+        self.file_path = file_path
+        self:readFile(file_path)
+    end
+    self.input = InputDialog:new{
+        title =  _("Basic text editor"),
+        input = self.context,
+        text_height = Screen:getHeight() * 0.5,
+        width = self.width or Screen:getWidth(),
+        input_type = "string",
+        enter_callback = function ()
+            return '\n'
```

Comment:
Nit

Reply:
what here?

Reply:
Single quotes instead of double quotes.

Reply:
I assume it's just for code style? doesn't change the logic/behavior? right?

If so I'll change it later as this code doesn't work anyway. 
see my comment for tests: https://github.com/koreader/koreader/pull/4070#issuecomment-404287434

Reply:
If a nitpick were to change behavior I don't think it'd be a nitpick anymore. ;-)